### DISCUSSION TITLE
Does a port of a Template Studio WinUI Windows app to an uno-platform app for Skia-GTK exist?

### DISCUSSION THREAD

**Discussion by derMaaster** [unoplatform/uno#11508] [Q&A]
Hi, 

I tried porting [this](https://github.com/derMaaster/TemplateStudioWinUI) template that was created on my Windows PC with Visual Studio 2022, and this is my attempt thus far [here](https://github.com/derMaaster/TempStudio_WinUI).

But I think it is a bit above my current skill level...     **does such a port or similar finished example exist?**

The template is an app with Navigation Pane + MVVM + Dependency Injection + some pages.

**Comment by jeromelaban** [comment]
Thanks for the question. We do not have such a template at this time.
You're mentioning:

> Commented out in ShellPage.xaml.cs: App.MainWindow.Activated += MainWindow_Activated; As well as the void MainWindo_Activated - Uno extension of and Microsoft.UI.Xaml.WindowActivatedEventArgs differs significantly it seems.

There's indeed a difference in startup that is related to our pending item of adding more support for multi-window APIs.

> NavigationViewHeaderBehavior.cs using Microsoft.Xaml.Interactivity - struggling to get the correct replacement

Could you detail a bit more this part?

**Comment by derMaaster** [comment]
On the part you ask to elaborate on - I managed to fix that in the meantime. I was struggling to get the equivalent package / namespace - Uno.CommunityToolkit.WinUI.UI.Behaviors fixed that.

Another comment on the ShellPage.xaml.cs errors - the public class Page that is inherited, it seems that the Uno Toolkit version is much more basic, and the difference between that and the original WinUI class is what causes the error.
